### PR TITLE
drivers/wireles/gs2200m: add NET_TCP dependency to NFS_DONT_BIND_TCP_SOCKET option

### PR DESCRIPTION
**Pull request by pkarashchenko**
## Summary
The NFS_DONT_BIND_TCP_SOCKET should be selected only TCP networking is enabled

## Impact
N/A

## Testing
Run `make menuconfig` and ensure the NFS options are correctly aligned.

### CODE REVIEW

File: fs/nfs/Kconfig [hunk 1]
```diff
@@ -17,6 +17,7 @@ config NFS_DONT_BIND_TCP_SOCKET
 	bool
 	default n
 	depends on NFS
```
Comment (patacongo) [inline fs/nfs/Kconfig:19]: 18    depends on NFS

Within

    13 if NFS

Is redundant, is it not?

Comment (pkarashchenko) [reply fs/nfs/Kconfig:19]: Seems to be true. Thanks for noticing!

Comment (pkarashchenko) [reply fs/nfs/Kconfig]: Fixed!

**Comment by pkarashchenko** [comment]
@masayuki2009 just a notice: the `kconfig-frontends` is used instead of `kconfiglib` when I try to `make menuconfig` after `./tools/configure.sh spresense:wifi_smp`

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by masayuki2009** [comment]
>@masayuki2009 just a notice: the kconfig-frontends is used instead of kconfiglib when I try to make menuconfig after ./tools/configure.sh spresense:wifi_smp. Do you know why?

@pkarashchenko 
I don't know the reason why but I've been using kconfig-frontends.

**Review by masayuki2009 — APPROVED** [review]

**Comment by pkarashchenko** [comment]
@xiaoxiang781216 any ideas about kconfig-frontends?

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
> @xiaoxiang781216 any ideas about kconfig-frontends?

The selection is done by checking menuconfig exist from kconfiglib:
https://github.com/apache/nuttx/blob/master/tools/Unix.mk#L616-L655

**Comment by pkarashchenko** [comment]
I have both installed and usually kconfiglib is used but in this particular case the kconfig-frontends is selected

**Comment by xiaoxiang781216** [comment]
If so, need add some log to debug:(

**Pull request merged**